x/stakeibc/keeper/msg_server_restore_interchain_account.go

```diff
@@ -44,15 +44,58 @@ func (k msgServer) RestoreInterchainAccount(goCtx context.Context, msg *types.Ms
 
 	// If we're restoring a delegation account, we also have to reset record state
 	if msg.AccountType == types.ICAAccountType_DELEGATION {
-		depositRecords := k.RecordsKeeper.GetAllDepositRecord(ctx)
 		// revert DELEGATION_IN_PROGRESS records for the closed ICA channel (so that they can be staked)
-		for _, record := range depositRecords {
+		depositRecords := k.RecordsKeeper.GetAllDepositRecord(ctx)
+		for _, depositRecord := range depositRecords {
+			// only revert records for the select host zone
+			if depositRecord.HostZoneId == hostZone.ChainId && depositRecord.Status == recordtypes.DepositRecord_DELEGATION_IN_PROGRESS {
+				depositRecord.Status = recordtypes.DepositRecord_DELEGATION_QUEUE
+				k.Logger(ctx).Info(fmt.Sprintf("Setting DepositRecord %d to status DepositRecord_DELEGATION_IN_PROGRESS", depositRecord.Id))
+				k.RecordsKeeper.SetDepositRecord(ctx, depositRecord)
+			}
+		}
+
+		// revert epoch unbonding records for the closed ICA channel
+		epochUnbondingRecords := k.RecordsKeeper.GetAllEpochUnbondingRecord(ctx)
+		recordsStuckInUnbonding := []uint64{}
```

Comment:
nit: should we call this `epochNumberForPendingRecord`  or something similar?

Reply:
yup, good call, updating